lxd/instance/drivers/driver_qemu.go

```diff
@@ -1029,6 +1029,13 @@ func (d *qemu) start(stateful bool, op *operationlock.InstanceOperation) error {
 		volatileSet["volatile.uuid"] = instUUID
 	}
 
+	// For a VM instance, we must also set the VM generation ID.
```

Comment:
This logic is instance type agnostic do I'm surprised to see it in the instance driver. If it has to be here, then its also missing from the lxc driver.

Reply:
This bit is at line 1969 in `driver_lxc.go`

```go
// For a container instance, we must also set the generation UUID.
genUUID := d.localConfig["volatile.uuid.generation"]
if genUUID != "" {
	genUUID = instUUID
	volatileSet["volatile.uuid.generation"] = genUUID
}
```

Reply:
OK, so no need for the "For <instance type>," bit in the comment, it suggests it only occurs for this specific instance type.

Reply:
Also, the logic looks wrong, as it should be `==` not `!=` right?

Suggest something like this:
```go
// Ensure generation UUID is populated.
if d.localConfig["volatile.uuid.generation"] == "" {
	d.localConfig["volatile.uuid.generation"] = instUUID
	volatileSet["volatile.uuid.generation"] = instUUID
}
```

Reply:
The logic here still looks wrong to me? Am i right?